lib/Db/SessionMapper.php

```diff
@@ -142,6 +142,42 @@ public function deleteInactiveWithoutSteps(?int $documentId = null): int {
 		return $deletedCount;
 	}
 
+	public function deleteOrphanedSteps(): int {
+		$startTime = microtime(true);
+		$maxExecutionSeconds = 30;
+		$batchSize = 1000;
+		$deletedCount = 0;
+
+		do {
+			$orphanedStepsBuilder = $this->db->getQueryBuilder();
+			$orphanedStepsBuilder->select('st.id')
+				->from('text_steps', 'st')
+				->leftJoin('st', 'text_sessions', 's', $orphanedStepsBuilder->expr()->eq('st.session_id', 's.id'))
```

Comment:
Ouch, indeed as @max-nextcloud pointed out, this should rather only regard steps that have no corresponding document anymore in the `text_documents` table.